File: leetcode/python/1578_minimum_time_to_make_rope_colorful.py
class Solution:
    def minCost(self, colors: str, neededTime: list[int]) -> int:
        res: int = 0
        i: int = 0
        while i < len(colors):
            curMax = total = neededTime[i]
            while i < len(colors) - 1 and colors[i] == colors[i+1]:
                i += 1
                total += neededTime[i]
                curMax = max(curMax, neededTime[i])
            i += 1
            res += total - curMax
        return res

s = Solution()
print(s.minCost(colors = "abaac", neededTime = [1,2,3,4,5]))
print(s.minCost(colors = "abc", neededTime = [1,2,3]))
print(s.minCost(colors = "aabaa", neededTime = [1,2,3,4,1]))

File: leetcode/rust/src/1578_minimum_time_to_make_rope_colorful.rs
impl Solution {
    pub fn min_cost(colors: String, needed_time: Vec<i32>) -> i32 {
        let mut res = 0;
        let mut i = 0;
        while i < colors.len() {
            let mut cur_max = 0;
            let color = &colors[i..i+1];

            while i < colors.len() && &colors[i..i+1] == color {
                cur_max = cur_max.max(needed_time[i]);
                res += needed_time[i];
                i += 1;
            }
            res -= cur_max;
        }
        res
    }
}